format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java

```diff
@@ -89,7 +90,12 @@ public Schema detectSchema(FormatContext context, InputFiles inputFiles) throws
         continue;
       }
       try (DataFileStream<GenericRecord> dataFileStream = new DataFileStream<>(inputFile.open(), datumReader)) {
-        return new AvroToStructuredTransformer().convertSchema(dataFileStream.getSchema());
+        Schema schema = new AvroToStructuredTransformer().convertSchema(dataFileStream.getSchema());
+        List<Schema.Field> fieldsList = schema.getFields();
+        if (conf.getPathField() != null) {
+          fieldsList.add(Schema.Field.of(conf.getPathField(), Schema.of(Schema.Type.STRING)));
+        }
+        return Schema.recordOf("Avro", fieldsList);
```

Comment:
use `schema.getRecordName()` as the schema name

Reply:
Changed to schema.getRecordName()